### PR TITLE
Finish implementing rxjs portion of the code.

### PR DESCRIPTION
**Pull request by jdfreder**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,5 +150,5 @@ Before opening a pull request, please run the unit tests locally:
 npm test
 ```
 
-You can also verify that the code works by hand by opening the test.html file
-in your web browser.
+You can also verify that the code works by hand by opening the `test.html` file
+in your web browser and following the promps.
```
Comment (rgbkrk) [inline README.md]: Prompts

File: src/index.js [hunk 2]
```diff
@@ -14,28 +19,55 @@ export function spawn(connectionOptions, kernelName) {
...
 export function disconnect(channels) {
-  channels.shell.disconnect();
-  channels.stdio.disconnect();
-  channels.iopub.disconnect();
-  channels.control.disconnect();
+  channels.shell.complete();
+  channels.stdio.complete();
+  channels.iopub.complete();
+  channels.control.complete();
+  return Promise.resolve();
```
Comment (rgbkrk) [inline src/index.js]: Is this future proofing for a later completion check?

File: src/index.js [hunk 3]
```diff
@@ -52,6 +84,129 @@ function _getKernelById(connectionOptions, kernelId) {
...
+/**
+ * Returns a websocket given a jupyter-js-services kernel
+ *
+ * This method creates a clone of the websocket stored inside the
+ * jupyter-js-services kernel instance.  It then quietly closes and disposes
+ * of the kernel object and its websocket.
+ * @param  {object} kernel kernel object returned from jupyter-js-services
+ * @return {WebSocket}     websocket
+ */
+function _stealKernelWebsocket(kernel) {
```
Comment (rgbkrk) [inline src/index.js]: Heh, do you only need the REST API from Jupyter js services then?

File: src/index.js [hunk 4]
```diff
@@ -52,6 +84,129 @@ function _getKernelById(connectionOptions, kernelId) {
...
+ * Create an observable from a websocket
+ * @param  {WebSocket} ws
+ * @return {Observable}
+ */
+function _wsObservable(ws) {
+  return new rxjs.Observable(subscriber => {
+    ws.onmessage = msg => {
+      try {
+        subscriber.next(msg);
+      } catch(error) { /* nom nom nom */ }
```
Comment (rgbkrk) [inline src/index.js]: If you catch an error here, it should subscriber.error instead (which should be handled by Rx if thrown anyway).

File: src/index.js [hunk 5]
```diff
@@ -52,6 +84,129 @@ function _getKernelById(connectionOptions, kernelId) {
...
+
+/**
+ * Create a channel Subject from a multiplexed Subject
+ * @param  {Subject} wsSubject
+ * @param  {string} channelName  Jupyter spec channel name
+ * @return {Subject}
+ */
+function _multiplexChannel(wsSubject, channelName) {
+  const subscriber = rxjs.Subscriber.create(
+    msg => {
```
Comment (rgbkrk) [inline src/index.js]: Nice multiplexing setup!

File: src/index.js (hunk 2 above)
Comment (jdfreder) [inline src/index.js]: :+1:

File: src/index.js (hunk 3 above)
Comment (jdfreder) [inline src/index.js]: Ya

**Pull request merged**